packages/process/src/node/terminal-process.ts

```diff
@@ -46,6 +52,29 @@ export class TerminalProcess extends Process {
 
         this.logger.debug('Starting terminal process', JSON.stringify(options, undefined, 2));
 
+        // node-pty doesn't handle the `shell` option by default.
+        // work-around: spawn the shell ourselves.
+        if (options.options && options.options.shell) {
+            let shell: string | undefined;
+            let shellArgs: string[] | undefined;
+
+            if (typeof options.options.shell === 'object') {
+                shell = options.options.shell.executable;
+                shellArgs = options.options.shell.args;
+            }
+
+            const command = options.command;
+            if (process.platform === 'win32') {
+                options.command = shell || 'cmd.exe';
+                options.args = shellArgs || ['/c'];
+                options.args.push(command);
+            } else {
+                options.command = shell || 'sh';
```

Comment:
"sh" is probably the way to go, but I wonder if some users might expect that the shell being used for shell processes would be the same one being used to spawn terminals (which is configurable, IIRC?

Reply:
nevermind - I think this code already does use the same shell we use for the terminal process.